### ISSUE TITLE
Unable to export CSV runtime data

### ISSUE THREAD

**Issue by LifeIsStrange** [material-table-core/core#579]
Hi @oze4 @mindaugasnakrosis friendly ping :)

I have a basic use case that is unfortunately not fulfilled by the library (it seems)
```
        {
            title: 'User ID',
            field: 'id'
        },
        {
            title: 'is userID present in new database',
            field: 'newDatabase',
            render: ({id}: Dataset) => {
                return <PresenceChecker id={id}/>
            }
```

I have used the standard exportCSV function and when I export,
the resulting file has the right column titles, and the first column show all the data, which is working as intented.
**however**, the second column does not shows any data.. the reason is that the field 'newDatabase' is empty, it's not what I want to export. I want to export what my users want, which is: the TEXT content of the render function !
To simplify, consider that the PresenceChecker component has simply a div that will show "found" or "not found" in a div, based on the response of a network call inside (if the userId is present in databaseX)

so basically what I want is to be able to export to CSV what the content that the user see.
to do that you would provide a new optional key in columns:
exportRuntimeValueAsString: true
or exportRenderContentAsString which you could achieve e.g. in native DOM via a .innerText

since many tables have some rows that have a result depending on subsequent network calls, in practice, your current exporter does not work for many people, including me.

related https://github.com/mbrn/material-table/issues/2594
unclear if related to https://github.com/oze4/material-table-core/issues/2

**Comment by Domino987** [comment]
Hi, simply parsing a jsx element to text is not possible and if the field is empty, you need to tell the table what the value is supposed to be.
Refer to this prop to get the correct value for the column: https://github.com/material-table-core/core/blob/master/types/index.d.ts#L197

fyi: this is not a bug

**Comment by LifeIsStrange** [comment]
Sorry for the delay,

> simply parsing a jsx element to text is not possible

That is irrelevant since you can do parsing using native DOM via .innerText

> Refer to this prop to get the correct value for the column: https://github.com/material-table-core/core/blob/master/types/index.d.ts#L197

Thank you, is there an example on how to implement the customExport prop?
